app/src/main/java/com/bogdan/codeforceswatcher/activity/AddUserActivity.kt

```diff
@@ -43,32 +43,37 @@ class AddUserActivity : AppCompatActivity(), OnClickListener {
                 if (response.isSuccessful) {
                     if (response.body()!!.result.firstOrNull() == null) {
                         progressBar.visibility = View.INVISIBLE
-                        showError()
+                        showError(getString(R.string.wrong))
                     } else {
                         val localUser = response.body()!!.result.firstOrNull()!!
                         ratingCall.enqueue(object : Callback<RatingChangeResponse> {
                             override fun onResponse(call: Call<RatingChangeResponse>, response: Response<RatingChangeResponse>) {
-                                progressBar.visibility = View.INVISIBLE
                                 if (response.isSuccessful) {
                                     localUser.ratingChanges = response.body()!!.result
                                     val findUser = CwApp.app.userDao.getAll().find { it.handle == localUser.handle }
                                     if (findUser == null) {
                                         CwApp.app.userDao.insert(localUser)
+                                        progressBar.visibility = View.INVISIBLE
```

Comment:
If you do `progressBar.visibility = View.INVISIBLE` in all if branches, why to not make it only once after all code executed?